src/test/scala/ru/d4team/chat/services/ChatRoomsServiceSpec.scala

```diff
@@ -0,0 +1,50 @@
+package ru.d4team.chat.services
+
+import ru.d4team.chat.services.ChatRoomsService._
+import ru.d4team.chat.utils._
+import zio._
+import zio.test._
+
+import scala.collection.concurrent.TrieMap
+import zio.test.TestAspect._
+
+object ChatRoomsServiceSpec extends ZIOSpecDefault {
+  override def spec: Spec[TestEnvironment with Scope, Any] = suite("ChatRoomsServiceSpec")(
+    test("'join' should create a new room for a first participant") {
+      check(Gen.alphaNumericString) { room =>
+        val testF = for {
+          // given
+          chatRooms    <- ZIO.service[ChatRoomsService[ChatRoomsMap]]
+          channel      <- TestWebSocketChannel.make
+          expectedRoom  = TrieMap(channel.id -> channel)
+          expectedRooms = TrieMap(room -> expectedRoom).asInstanceOf[ChatRoomsMap]
```

Comment:
If we dont need fields from `TestWebSocketChannel`, we can change the contract of method `make` to `def make: ZIO[Any, Nothing, Channel[WebSocketFrame]]` and remove unsave `asInstanceOf`

Reply:
Good point, thank you